### PR TITLE
feat: list and view templates

### PR DESCRIPTION
**Pull request by JohnXander**

### CODE REVIEW

File: src/components/lessons/LessonView.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import ViewTemplate from '../../utils/templates/ViewTemplate'
+import { useLocation } from 'react-router-dom'
+
+const LessonView = () => {
+    const location = useLocation()
+
+    return (
+        <>
```
Comment (Auenc) [inline src/components/lessons/LessonView.js:8]: I don't think you need the fragment here, just return the ViewTemplate

File: src/components/module/ModuleList.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        learningObj: ['Objective A', 'Objective B']
+    }
+]
+
+const templateType = 'module'
+
+const ModuleList = () => {
+    return (
+        <>
+            <ListTemplate
```
Comment (Auenc) [inline src/components/module/ModuleList.js:23]: Same as above, fragment not needed

File: src/components/module/ModuleView.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import ViewTemplate from '../../utils/templates/ViewTemplate'
+import { useLocation } from 'react-router-dom'
+
+const ModuleView = () => {
+    const location = useLocation()
+
+    return (
+        <>
```
Comment (Auenc) [inline src/components/module/ModuleView.js:8]: Same as above

File: src/components/unit/UnitList.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        learningObj: ['React Objective A', 'React Objective B']
+    }
+]
+
+const templateType = 'unit'
+
+const UnitList = () => {
+    return (
+        <>
+            <ListTemplate
```
Comment (Auenc) [inline src/components/unit/UnitList.js:23]: Same as above

File: src/components/unit/UnitView.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import ViewTemplate from '../../utils/templates/ViewTemplate'
+import { useLocation } from 'react-router-dom'
+
+const UnitView = () => {
+    const location = useLocation()
+
+    return (
+        <>
```
Comment (Auenc) [inline src/components/unit/UnitView.js:8]: Same as above

File: src/utils/templates/ListTemplate.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Button } from '@mui/material';
+import './style.css'
+import { useNavigate } from 'react-router-dom';
+
+const ListTemplate = ({ templateData, templateType }) => {
+    const templateCode = templateType.charCodeAt(0)
```
Comment (Auenc) [inline src/utils/templates/ListTemplate.js:6]: Would be good to check if `templateType` isn't null/or an empty string before using it

**Review by Auenc — CHANGES_REQUESTED** [review]

**Review by mnmin — APPROVED** [review]

**Review by mnmin — APPROVED** [review]

**Pull request merged**